internal/cli/flags.go

```diff
@@ -15,6 +15,15 @@ type Flag struct {
 	IsRequired bool
 }
 
+// Required clones an existing flag and assigns `IsRequired` to true. This is
+// useful when there are flags with several different use cases -- e.g. create
+// requiring the flag, but update not requiring it.
+func (f *Flag) Required() *Flag {
+	clone := *f
+	clone.IsRequired = true
+	return &clone
+}
+
```

Comment:
@jfatta @Widcket While iterating on this rules feature, the difference between create requiring a name and update not -- made me first repeat the same struct.

This is an approach to DRY up that. Lemme know what you think :)